### PR TITLE
Development and cleanup

### PR DESCRIPTION
**Pull request by tablatronix**
cleaning up a bit, better comments, some questions `@todo`, improved some debug messages.
abstracted appassword check, and made fatal, not fallback to anonymous.

### CODE REVIEW

File: WiFiManager.cpp [hunk 1]
```diff
@@ -91,36 +94,29 @@ void WiFiManager::setupConfigPortal() {
...
 
+  // start soft AP with pass or anonymous
   if (_apPassword != NULL) {
     WiFi.softAP(_apName, _apPassword);//password option
   } else {
     WiFi.softAP(_apName);
   }
 
-  delay(500); // Without delay I've seen the IP address blank
+  delay(500); // slight delay to make sure we have a password
```
Comment (tablatronix) [inline WiFiManager.cpp]: I meant ip address doh

**Comment by tablatronix** [comment]
hate it when github updates the site when I am using it, heh looks all different now.

**Comment by tablatronix** [comment]
I am currently trying to figure out this one resilient problem, when the configportal save fails, and the portal is closed or timesout, somehow, persistent mode is being set to AP or AP_STA which ever we are using, and i cannot for the life of me get it to turn off, ever using custom code that should. I have opened an issue with arduino sdk and are looking into the SDK bugs to see if this is a known issue fixed in 2.0, ill test master soon. basically softap disconnect fails to save config to flash it seems. Even though I am in AP mode as per the SDK, there seems to be some bug here, still narrowing it down .

**Comment by tzapu** [comment]
maybe the not saving issue could be caused by some combination of the persistent flags and switching modes? wouldn't be the weirdest thing i ve seen it do...

**Comment by tablatronix** [comment]
I spend all weekend on this and have it figured out. Ill detail it later, I also might have a fix.

**Comment by tablatronix** [comment]
Sorry I dropped dev on this, I have been having problems getting around to programming projects.

**Comment by tzapu** [comment]
don't worry mate, same here, do you want me to merge it into dev?

**Comment by tablatronix** [comment]
Im gonna work on it some more fix the 2 major issues we have with stuck stuff

**Comment by tablatronix** [comment]
@tzapu can I merge and continue work on this ?

**Comment by tzapu** [comment]
please do, you are the lead my man

thank you

**Pull request merged**